ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/builder/BuilderApi.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.blockcreation.builder;
+
+import org.hyperledger.besu.datatypes.BLSPublicKey;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.core.BlockBody;
+import org.hyperledger.besu.ethereum.core.ExecutionPayloadResponse;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Builder API service */
+public class BuilderApi {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BuilderApi.class);
+
+  OkHttpClient client;
+  public String endpoint;
+
+  public BLSPublicKey publicKey;
+
+  public BuilderApi(final String endpoint, final BLSPublicKey blsPublicKey) {
```

Comment:
This is (builder) API client, not an API (naming)
Consider renaming to `BuilderClient` or `BuilderAPIClient` 🤔